File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        res = 0
        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            if nums[index] < 0:
                return index + 1
            nums[index] *= -1
        return 0

File: Data Structures & Algorithms/concatenation-of-array/submission-2.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len();
        let mut res = vec![0; n*2];
        for i in 0..n {
            res[i] = nums[i];
            res[i+n] = nums[i];
        }
        res 
    }
}
